xdcr/xdcr_test.go

```diff
@@ -50,74 +52,299 @@ func TestMobileXDCRNoSyncDataCopied(t *testing.T) {
-		for _, doc := range []string{normalDoc, attachmentDoc} {
-			require.EventuallyWithT(t, func(c *assert.CollectT) {
-				var value string
-				_, err = toDs.Get(doc, &value)
-				assert.NoError(c, err, "Could not get doc %s", doc)
-				assert.Equal(c, body, value)
-			}, time.Second*5, time.Millisecond*100)
-		}
+	// stats are not updated in real time, so we need to wait a bit
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		stats, err := xdcr.Stats(ctx)
+		assert.NoError(t, err)
+		assert.Equal(c, totalDocsFiltered+1, stats.DocsFiltered)
+		assert.Equal(c, totalDocsWritten+2, stats.DocsWritten)
 
-		var value any
-		_, err = toDs.Get(syncDoc, &value)
-		base.RequireDocNotFoundError(t, err)
+	}, time.Second*5, time.Millisecond*100)
+}
 
-		// stats are not updated in real time, so we need to wait a bit
-		require.EventuallyWithT(t, func(c *assert.CollectT) {
-			stats, err := xdcr.Stats(ctx)
-			assert.NoError(t, err)
-			assert.Equal(c, totalDocsFiltered+1, stats.DocsFiltered)
-			assert.Equal(c, totalDocsWritten+2, stats.DocsWritten)
+// getTwoBucketDataStores creates two data stores in separate buckets to run xdcr within. Returns a named collection or a default collection based on the global test configuration.
+func getTwoBucketDataStores(t *testing.T) (base.Bucket, sgbucket.DataStore, base.Bucket, sgbucket.DataStore) {
+	ctx := base.TestCtx(t)
+	base.RequireNumTestBuckets(t, 2)
+	fromBucket := base.GetTestBucket(t)
+	t.Cleanup(func() {
+		fromBucket.Close(ctx)
+	})
+	toBucket := base.GetTestBucket(t)
+	t.Cleanup(func() {
+		toBucket.Close(ctx)
+	})
+	var fromDs base.DataStore
+	var toDs base.DataStore
+	if base.TestsUseNamedCollections() {
+		var err error
+		fromDs, err = fromBucket.GetNamedDataStore(0)
+		require.NoError(t, err)
+		toDs, err = toBucket.GetNamedDataStore(0)
+		require.NoError(t, err)
+	} else {
+		fromDs = fromBucket.DefaultDataStore()
+		toDs = toBucket.DefaultDataStore()
+	}
+	return fromBucket, fromDs, toBucket, toDs
+}
 
-		}, time.Second*5, time.Millisecond*100)
-		totalDocsWritten += 2
-		totalDocsFiltered++
-		if base.UnitTestUrlIsWalrus() {
-			// TODO: CBG-3861 implement _vv support in rosmar
-			continue
-		}
-		// in mobile xdcr mode a version vector will be written
-		if base.TestSupportsMobileXDCR() {
-			// verify VV is written to docs that are replicated
-			for _, doc := range []string{normalDoc, attachmentDoc} {
-				require.EventuallyWithT(t, func(c *assert.CollectT) {
-					xattrs, _, err := toDs.GetXattrs(ctx, doc, []string{"_vv"})
-					assert.NoError(c, err, "Could not get doc %s", doc)
-					vvXattrBytes, ok := xattrs["_vv"]
-					require.True(t, ok)
-					var vvXattrVal map[string]any
-					require.NoError(t, base.JSONUnmarshal(vvXattrBytes, &vvXattrVal))
-					assert.NotNil(c, vvXattrVal[version])
-					assert.NotNil(c, vvXattrVal[source])
-					assert.NotNil(c, vvXattrVal[curCAS])
-
-				}, time.Second*5, time.Millisecond*100)
-			}
-		}
+func TestReplicateVV(t *testing.T) {
```

Comment:
This test looks like what's it's actually testing is XDCR's LWW logic?  Should it be renamed TestReplicateLWW?